File: cmd/yeehaw/scheduler.go
package main

import (
	"fmt"

	"github.com/spf13/cobra"
)

var schedulerCmd = &cobra.Command{
	Use:   "scheduler",
	Short: "Manage scheduler configuration",
}

var schedMaxGlobal int
var schedMaxProject int
var schedTimeout int

var schedulerConfigCmd = &cobra.Command{
	Use:   "config",
	Short: "View or update scheduler limits",
	Long:  "Without flags, shows current config. With flags, updates the specified values.",
	RunE: func(cmd *cobra.Command, args []string) error {
		db := mustOpenDB()
		defer db.Close()

		// Check if any flags were explicitly set
		globalSet := cmd.Flags().Changed("max-global")
		projectSet := cmd.Flags().Changed("max-project")
		timeoutSet := cmd.Flags().Changed("timeout")

		if globalSet || projectSet || timeoutSet {
			var gp, pp, tp *int
			if globalSet {
				gp = &schedMaxGlobal
			}
			if projectSet {
				pp = &schedMaxProject
			}
			if timeoutSet {
				tp = &schedTimeout
			}
			if err := db.UpdateSchedulerConfig(gp, pp, tp); err != nil {
				return err
			}
			fmt.Println("Scheduler config updated.")
		}

		cfg, err := db.GetSchedulerConfig()
		if err != nil {
			return err
		}

		fmt.Printf("Scheduler Configuration:\n")
		fmt.Printf("  Max global tasks:      %d\n", cfg.MaxGlobal)
		fmt.Printf("  Max per-project tasks: %d\n", cfg.MaxPerProject)
		fmt.Printf("  Timeout (minutes):     %d\n", cfg.TimeoutMinutes)
		return nil
	},
}

func init() {
	schedulerConfigCmd.Flags().IntVar(&schedMaxGlobal, "max-global", 5, "maximum concurrent tasks globally")
	schedulerConfigCmd.Flags().IntVar(&schedMaxProject, "max-project", 3, "maximum concurrent tasks per project")
	schedulerConfigCmd.Flags().IntVar(&schedTimeout, "timeout", 60, "task timeout in minutes")

	schedulerCmd.AddCommand(schedulerConfigCmd)
}

File: src/yeehaw/cli/scheduler.py
"""Scheduler configuration commands."""

from __future__ import annotations

import argparse

from pathlib import Path
from typing import Any

from yeehaw.store.store import Store


def handle_scheduler(args: argparse.Namespace, db_path: Path) -> None:
    """Handle scheduler config show/update commands."""
    store = Store(db_path)
    try:
        if args.scheduler_command == "show":
            config = store.get_scheduler_config()
            print("Scheduler Configuration:")
            print(f"  Max global tasks:   {config['max_global_tasks']}")
            print(f"  Max per project:    {config['max_per_project']}")
            print(f"  Tick interval:      {config['tick_interval_sec']}s")
            print(f"  Task timeout:       {config['task_timeout_min']} min")

        elif args.scheduler_command == "config":
            updates: dict[str, int] = {}
            if args.max_global is not None:
                updates["max_global_tasks"] = args.max_global
            if args.max_project is not None:
                updates["max_per_project"] = args.max_project
            if args.tick is not None:
                updates["tick_interval_sec"] = args.tick
            if args.timeout is not None:
                updates["task_timeout_min"] = args.timeout

            if not updates:
                print("No changes specified.")
                return

            store.update_scheduler_config(**updates)
            print("Scheduler config updated:")
            for key, value in updates.items():
                print(f"  {key} = {value}")
    finally:
        store.close()
